testsuite/MDAnalysisTests/analysis/test_encore.py

```diff
@@ -639,114 +600,107 @@ def setUp(self):
         np.maximum(distances, 0, out=distances)
         distances.flat[::distances.shape[0] + 1] = 0.0
         dimension = len(distances)
-        self.distance_matrix = encore.utils.TriangularMatrix(len(distances))
+
+        distance_matrix = encore.utils.TriangularMatrix(len(distances))
         for i in range(dimension):
-            for j in range(i,dimension):
-                self.distance_matrix[i, j] = distances[i,j]
+            for j in range(i, dimension):
+                distance_matrix[i, j] = distances[i, j]
+        return distance_matrix
 
-    def test_one(self):
-        preference = -float(np.median(self.distance_matrix.as_array()) * 10.)
+    def test_one(self, distance_matrix):
+        preference = -float(np.median(distance_matrix.as_array()) * 10.)
         clustering_method = encore.AffinityPropagationNative(preference=preference)
         ccs = encore.cluster(None,
-                             distance_matrix=self.distance_matrix,
+                             distance_matrix=distance_matrix,
                              method=clustering_method)
-        assert_equal(self.n_clusters, len(ccs),
-                     err_msg="Basic clustering test failed to give the right"
-                             "number of clusters: {0} vs {1}".format(self.n_clusters, len(ccs)))
-
-
-class TestEncoreDimensionalityReduction(TestCase):
-    def setUp(self):
-        # Create universe from templates defined in setUpClass
-        self.ens1 = mda.Universe(
-            self.ens1_template.filename,
-            self.ens1_template.trajectory.timeseries(format='fac'),
+        assert self.n_clusters == len(ccs), \
+                     "Basic clustering test failed to give the right"\
+                    "number of clusters: {0} vs {1}".format(self.n_clusters, len(ccs))
+
+
+class TestEncoreDimensionalityReduction(object):
+    @pytest.fixture(scope='class')
+    def ens1_template(self):
+        template = mda.Universe(PSF, DCD)
+        template.transfer_to_memory(step=5)
+        template = mda.Universe(
+            template.filename,
+            np.copy(template.trajectory.timeseries(format='fac')),
```

Comment:
Do you know what this statement is doing?  After using transfer to memory this shouldn't be necessary anymore.

Reply:
I don't think I have a real understanding of that statement.